client/client-lib/src/lib.rs

```diff
@@ -554,14 +555,12 @@ impl<T: AsRef<ClientConfig> + Clone> Client<T> {
             let (mut keys, input) = MintClient::ecash_input(notes)?;
             tx.input(&mut keys, input);
             self.submit_tx_with_change(tx, rng).await?;
-            self.fetch_all_notes().await;
+            self.fetch_all_notes().await?;
             self.mint_client().select_notes(amount).await?
         };
-        assert_eq!(
-            final_notes.total_amount(),
-            amount,
-            "should have exact change"
-        );
+        if final_notes.total_amount() != amount {
+            return Err(ClientError::SpendReusedNote);
```

Comment:
I don't get how this is the appropriate error here, might be useful to add a comment? How would `SpendReuseNote` lead to `final_notes.total_amount() != amount`?

Reply:
If we fetched all the notes the only reason we wouldn't have enough at this point is if they were spent concurrently.

This is the failure that @justinmoon saw in the wild.